sources/packages/web/src/components/common/restriction/ViewRestriction.vue

```diff
@@ -73,19 +73,14 @@
         </template>
       </template>
       <template #footer>
-        <check-permission-role :role="allowedRole">
-          <template #="{ notAllowed }">
-            <footer-buttons
-              :processing="processing"
-              :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
-              secondaryLabel="Cancel"
-              @primaryClick="allowUserToEdit ? submit() : cancel()"
-              @secondaryClick="cancel"
-              :disablePrimaryButton="notAllowed"
-              :showSecondaryButton="allowUserToEdit"
-            />
-          </template>
-        </check-permission-role>
+        <footer-buttons
+          :processing="processing"
+          :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
```

Comment:
I would suggest the following quick solution which would keep the primary behavior as primary and secondary behaviour as secondary not having to juggle based on the situation.

**Solution**

```vue
<template #footer>
        <check-permission-role :role="allowedRole">
          <template #="{ notAllowed }">
            <footer-buttons
              :processing="processing"
              primaryLabel="Resolve restriction"
              :secondaryLabel="allowUserToEdit ? 'Cancel' : 'Close'"
              @primaryClick="submit"
              @secondaryClick="cancel"
              :disablePrimaryButton="notAllowed"
              :showPrimaryButton="allowUserToEdit"
            />
          </template>
        </check-permission-role>
      </template>
```

Result: 

View Active Restriction with permission

![image](https://github.com/user-attachments/assets/b4435878-c514-492a-a95e-1049e5792432)


View resolved restriction with permission

![image](https://github.com/user-attachments/assets/402d3370-7ec5-462d-b80d-ae58de75b575)

View Resolved restriction without permission

![image](https://github.com/user-attachments/assets/5b695fa8-939a-4396-8247-954ce11865cb)

Reply:
The solution looks great, and we can probably adopt this in other stories. Thanks @dheepak-aot